### ISSUE TITLE
ICE: Opaque types got hidden types registered from within subtype predicate

### ISSUE THREAD

**Issue by msfjarvis** [rust-lang/rust#125516]
### Code

https://github.com/teloxide/teloxide/blob/bd98ba7e613e0fd7ec4fd1c210fc5edbb048f975/crates/teloxide-core/src/types/update.rs#L195

I wasn't able to create a minimal reproducer :(

Here's the source code of the project where the crash is triggered: https://github.com/msfjarvis/linkleaner/tree/1a327237a91f80fcead1826d4fc916704ed6a027

### Meta

`rustc --version --verbose`:
```
rustc 1.80.0-nightly (867900499 2024-05-23)
binary: rustc
commit-hash: 8679004993f08807289911d9f400f4ac4391d2bc
commit-date: 2024-05-23
host: x86_64-unknown-linux-gnu
release: 1.80.0-nightly
LLVM version: 18.1.6
```

The last `rustc` build that this project was using is `nightly-2024-01-09`, where this crash does not happen.

### Error output

```
error: internal compiler error: compiler/rustc_infer/src/infer/mod.rs:968:17: opaque types got hidden types registered from within subtype predicate: either::Either<either::Either<Alias(Opaque, AliasTy { args: ['?50], def_id: DefId(0:32657 ~ teloxide_core[4d4f]::types::poll::{impl#0}::mentioned_users::{opaque#0}) }), ?200t>, ?201t> vs either::Either<either::Either<Alias(Opaque, AliasTy { args: ['?50], def_id: DefId(0:32657 ~ teloxide_core[4d4f]::types::poll::{impl#0}::mentioned_users::{opaque#0}) }), ?204t>, ?205t>
   --> /home/msfjarvis/.cargo/git/checkouts/teloxide-cd43b5216a237e6a/bd98ba7e613e/crates/teloxide-core/src/types/update.rs:195:54
    |
195 |             UpdateKind::ChosenInlineResult(query) => i1(once(&query.from)),
    |                                                      ^^^^^^^^^^^^^^^^^^^^^

```

</p>
</details>

**Comment by lqd** [comment]
Should be fixed by #125489 in tomorrow's nightly

**Comment by msfjarvis** [comment]
Confirmed fixed.
